src/main/java/org/opensearch/security/http/OnBehalfOfAuthenticator.java

```diff
@@ -0,0 +1,254 @@
+
+    protected final Logger log = LogManager.getLogger(this.getClass());
+
+    private static final Pattern BEARER = Pattern.compile("^\\s*Bearer\\s.*", Pattern.CASE_INSENSITIVE);
+    private static final String BEARER_PREFIX = "bearer ";
+    private static final String TOKEN_TYPE_CLAIM = "typ";
+    private static final String TOKEN_TYPE = "obo";
+
+    private final JwtParser jwtParser;
+    private final String encryptionKey;
+    private final Boolean oboEnabled;
+    private final String clusterName;
+
+    public OnBehalfOfAuthenticator(Settings settings, String clusterName) {
+        String oboEnabledSetting = settings.get("enabled");
+        oboEnabled = oboEnabledSetting == null ? Boolean.TRUE : Boolean.valueOf(oboEnabledSetting);
+        encryptionKey = settings.get("encryption_key");
+        jwtParser = initParser(settings.get("signing_key"));
+        this.clusterName = clusterName;
+    }
+
+    private JwtParser initParser(final String signingKey) {
+        JwtParser _jwtParser = keyUtil.createJwtParserFromSigningKey(signingKey, log);
+
+        if (_jwtParser == null) {
+            throw new RuntimeException("Unable to find on behalf of authenticator signing key");
+        }
+
+        return _jwtParser;
+    }
+
+    private List<String> extractSecurityRolesFromClaims(Claims claims) {
+        Object rolesObject = ObjectUtils.firstNonNull(claims.get("er"), claims.get("dr"));
+        List<String> roles;
+
+        if (rolesObject == null) {
+            log.warn("This is a malformed On-behalf-of Token");
+            roles = List.of();
+        } else {
+            final String rolesClaim = rolesObject.toString();
+
+            // Extracting roles based on the compatbility mode
+            String decryptedRoles = rolesClaim;
+            if (rolesObject == claims.get("er")) {
+                decryptedRoles = EncryptionDecryptionUtil.decrypt(encryptionKey, rolesClaim);
+            }
+            roles = Arrays.stream(decryptedRoles.split(",")).map(String::trim).collect(Collectors.toList());
+        }
+
+        return roles;
+    }
+
+    private String[] extractBackendRolesFromClaims(Claims claims) {
+        // Object backendRolesObject = ObjectUtils.firstNonNull(claims.get("ebr"), claims.get("dbr"));
+        if (!claims.containsKey("br")) {
+            return null;
+        }
+
+        Object backendRolesObject = claims.get("br");
+        String[] backendRoles;
+
+        if (backendRolesObject == null) {
+            log.warn("This is a malformed On-behalf-of Token");
+            backendRoles = new String[0];
+        } else {
+            final String backendRolesClaim = backendRolesObject.toString();
+
+            // Extracting roles based on the compatibility mode
+            String decryptedBackendRoles = backendRolesClaim;
+            backendRoles = Arrays.stream(decryptedBackendRoles.split(",")).map(String::trim).toArray(String[]::new);
+        }
```

Comment:
Seems like containsKey check is redundant. You can 
```suggestion
        Object backendRolesObject = claims.get("br");
        String[] backendRoles = null;
        if (backendRolesObject == null) {
            log.warn("This is a malformed On-behalf-of Token");
        } else {
            // Extracting roles based on the compatibility mode
            backendRoles = Arrays.stream(backendRolesObject.toString().split(",")).map(String::trim).toArray(String[]::new);
        }
```